extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.settings.js

```diff
@@ -0,0 +1,168 @@
+		toggleLabel.setAttribute('for', toggleID);
+		toggleLabel.classList.add('wds-switch__label');
+		toggleLabel.appendChild(document.createTextNode("Autoplay Videos"));
+		toggleLabel.addEventListener('click', function (event) {
+			featuredVideoAutoplay.toggleAutoplay(!event.target.previousSibling.checked);
+			playerInstance.trigger('autoplayToggle', !event.target.previousSibling.checked);
+		});
+
+		autoplayToggle.appendChild(toggleInput);
+		autoplayToggle.appendChild(toggleLabel);
+
+		return autoplayToggle;
+	};
+
+	WikiaJWPlayerSettings.prototype.createQualityLevelsList = function () {
+		var qualityLevelsList = document.createElement('ul'),
+			backButton = document.createElement('li'),
+			playerInstance = this.player,
+			isActiveClass = 'is-active',
+			self = this;
+		qualityLevelsList.classList.add('wikia-jw-settings__quality-levels');
+		backButton.classList.add('wikia-jw-settings__back');
+		backButton.innerHTML = '<svg class="wikia-jw-settings__back-icon" width="18" height="18" viewBox="0 0 18 18" data-reactid=".0.0.5.3.1.$quality.0.0.0.0.1.0"><path d="M9 14a.997.997 0 0 1-.707-.293l-7-7a.999.999 0 1 1 1.414-1.414L9 11.586l6.293-6.293a.999.999 0 1 1 1.414 1.414l-7 7A.997.997 0 0 1 9 14" data-reactid=".0.0.5.3.1.$quality.0.0.0.0.1.0.0"></path></svg> Back';
+		backButton.addEventListener('click', function () {
+			self.showSettingsList();
+		});
+		qualityLevelsList.appendChild(backButton);
+
+		playerInstance.on('levels', function (data) {
+			self.updateQualityLevelsList(qualityLevelsList, data.levels, isActiveClass, backButton);
+		});
+
+		playerInstance.on('levelsChanged', function (data) {
+			self.updateCurrentQuality(qualityLevelsList, data.currentQuality, isActiveClass);
+		});
+		return qualityLevelsList;
+	};
+
+	WikiaJWPlayerSettings.prototype.updateQualityLevelsList = function (qualityLevelsList, newLevels, isActiveClass, backButton) {
+		var playerInstance = this.player,
+			self = this;
+		while (qualityLevelsList.childElementCount > 1) {
+			qualityLevelsList.removeChild(qualityLevelsList.firstChild);
+		}
+		newLevels.forEach(function (level, index) {
+			var qualityLevelItem = document.createElement('li');
+			qualityLevelItem.addEventListener('click', function () {
+				playerInstance.setCurrentQuality(index);
+				self.close();
+			});
+			if (playerInstance.getCurrentQuality() === index) {
+				qualityLevelItem.classList.add(isActiveClass);
+			}
+			qualityLevelItem.appendChild(document.createTextNode(level.label));
+			qualityLevelsList.insertBefore(qualityLevelItem, backButton);
+		});
+	};
+
+	WikiaJWPlayerSettings.prototype.updateCurrentQuality = function (qualityLevelsList, currentQuality, isActiveClass) {
+		qualityLevelsList.childNodes.forEach(function (node, index) {
+			if (currentQuality === index) {
+				node.classList.add(isActiveClass);
+			} else {
+				node.classList.remove(isActiveClass);
+			}
+		});
+	};
+
+	WikiaJWPlayerSettings.addToPlayer = function () {
+		new WikiaJWPlayerSettings(this);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either remove this useless object instantiation of "WikiaJWPlayerSettings" or use it [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AConstructorFunctionsForSideEffects)